azurelinuxagent/common/protocol/hostplugin.py

```diff
@@ -266,8 +276,13 @@ def _base64_encode(self, data):
             return s.decode('utf-8')
         return s
     
-    def _read_response_error(self, response):
+    @staticmethod
+    def read_response_error(response):
+        if response is None:
+            return ''
         body = remove_bom(response.read())
+        if body is None:
+            return ''
```

Comment:
This change loses the status / reason when the body is absent. Do we want that? It seems they may be useful for other errors.